src/app/features/worklog/worklog-export/worklog-export.util.ts

```diff
@@ -152,40 +49,130 @@ export const createRows = (tasks: WorklogTask[], startTimes: WorkStartEnd, endTi
+          projects: (task.projectId) ? [task.projectId] : [],
+          tags: task.tagIds,
+          workStart: data.workTimes.start[day],
+          workEnd: data.workTimes.end[day],
+          timeSpent: 0,
+          timeEstimate: 0
+        };
+        if (!task.subTaskIds || task.subTaskIds.length === 0) {
+          taskGroups[day].timeSpent = task.timeSpentOnDay[day];
+          taskGroups[day].timeEstimate = task.timeEstimate / numDays;
+        }
+      });
+      break;
+    case WorklogGrouping.PARENT:
+    case WorklogGrouping.TASK:
+      taskGroups[task.id] = createEmptyGroup();
+
+      // by design subtasks don't have tags, so we must set its parent's tags
+      if (task.parentId !== null) {
+        taskGroups[task.id].tags = (data.tasks.find(t => t.id === task.parentId) as WorklogTask).tagIds;
+      } else {
+        taskGroups[task.id].tags = task.tagIds;
+      }
+
+      taskGroups[task.id].tasks = [task];
+      taskGroups[task.id].notes = (task.notes) ? [task.notes] : [];
+      taskGroups[task.id].projects = (task.projectId) ? [task.projectId] : [];
+      taskGroups[task.id].dates = Object.keys(task.timeSpentOnDay);
+      taskGroups[task.id].timeEstimate = task.timeEstimate;
+      taskGroups[task.id].timeSpent = Object.values(task.timeSpentOnDay).reduce((acc, curr) => acc + curr, 0);
+      break;
+    default: // group by work log (don't group at all)
+      Object.keys(task.timeSpentOnDay).forEach(day => {
+        const groupKey = task.id + '_' + day;
+        taskGroups[groupKey] = createEmptyGroup();
+        taskGroups[groupKey].tasks = [task];
+        taskGroups[groupKey].notes = (task.notes) ? [task.notes] : [];
+        taskGroups[groupKey].projects = (task.projectId) ? [task.projectId] : [];
+        taskGroups[groupKey].tags = task.tagIds;
+        taskGroups[groupKey].dates = [day];
+        taskGroups[groupKey].workStart = data.workTimes.start[day];
+        taskGroups[groupKey].workEnd = data.workTimes.end[day];
+        taskGroups[groupKey].timeEstimate = task.subTaskIds.length > 0 ? 0 : task.timeEstimate;
+        taskGroups[groupKey].timeSpent = task.subTaskIds.length > 0 ? 0 : task.timeSpentOnDay[day];
+      });
+  }
+  return taskGroups;
+};
+
+const createEmptyGroup = (): RowItem => {
+  return {
+    dates: [],
+    timeSpent: 0,
+    timeEstimate: 0,
+    tasks: [],
+    titlesWithSub: [],
+    titles: [],
+    notes: [],
+    projects: [],
+    tags: [],
+    workStart: undefined,
+    workEnd: undefined,
+  };
+};
+
+const sortDateStrings = (dates: string[]): string[] => {
+  return dates.sort((a: string, b: string) => {
+    const dateA: number = new Date(a).getTime();
+    const dateB: number = new Date(b).getTime();
+    if (dateA === dateB) {
+      return 0;
+    } else if (dateA < dateB) {
+      return -1;
+    }
+    return 1;
+  });
 };
 
 export const formatRows = (rows: RowItem[], options: WorklogExportSettingsCopy): (string | number | undefined)[][] => {
```

Comment:
I would love to see a short introductory comment on what this does and how, because I am pretty sure I will quickly forget this :)